pycheribuild/projects/cross/freertos.py

```diff
@@ -76,22 +94,24 @@ def __init__(self, config: CheriConfig):
         # Override that with no and set the appopriate flags here.
         self.make_args.set(NIX_ENV="no")
 
-        # Only build 64-bit FreeRTOS as cheribuild currently only supports building
-        # for RV64
-        self.make_args.set(RISCV_XLEN="64")
-
         # Set sysroot Makefile arg to pick up libc
         self.make_args.set(SYSROOT=str(self.sdk_sysroot))
 
         # Add compiler-rt location to the search path
         # self.make_args.set(LDFLAGS="-L"+str(self.compiler_resource / "lib"))
 
-        if self.target_info.target.is_cheri_purecap():
-            # CHERI-RISC-V sophisticated Demo with more advanced device drivers
-            # and currently only runs on FPGA-GFE, purecap
-            self.supported_freertos_demos.append("RISC-V_Galois_P1")
-            self.supported_demo_apps["RISC-V_Galois_P1"] = ["main_blinky", "main_netboot"]
+        if self.demo == "RISC-V_Galois_P1":
+            if "rv32" in self.demo_bsp:
```

Comment:
It seems to me like XARCH should only depend on whether the `crosscompile_target` is rv32 or rv64. Requesting one that doesn't match should just cause a `fatal_error`.